### DISCUSSION TITLE
How to specify initial state distribution when using stepthrough

### DISCUSSION THREAD

**Discussion by gdaddi** [JuliaPOMDP/POMDPs.jl#468] [Problem Implementation] [answered]
Hi,

I am trying to figure out from Julia POMDP's documentation how to specify an initial state distribution when running stepthrough simulations. But I can not find a good example and the documentation is not very clear.

I am trying this:
```
initial_state_dist = Deterministic( ... ) 
for (a, b, s) in stepthrough(planner, pomdp_problem, initial_state, max_steps=50)
    ...
end
```
I tried a bunch of permutations, but always seem to get some variation of `no method matching` for the stepthrough line. 

What is the right way of doing this?

**Comment by zsunberg** [comment, marked as answer]
The docs are here https://juliapomdp.github.io/POMDPs.jl/stable/POMDPTools/simulators/#POMDPTools.Simulators.stepthrough, but there are a lot of options, so it can be confusing.

If it is an MDP, ust sample from the distribution before calling stepthrough:
```julia
initial_state_dist = Deterministic( ... ) 
initial_state = rand(initial_state_dist)
for (a, b, s) in stepthrough(mdp, planner, initial_state, "a,b,s", max_steps=50)
    ...
end
```
If it is a POMDP, you have to explicitly pass the updater, and then the initial state distribution and initial belief are the same thing.
```julia
initial_state_dist = Deterministic( ... ) 
up = updater(planner)
for (a, b, s) in stepthrough(pomdp, planner, up, initial_state_dist, "a,b,s", max_steps=50)
    ...
end
```
